### PR TITLE
Added content guidance for Swagger devs around writing the descriptions that are used for docs.microsoft.com ref pages

### PR DESCRIPTION
**Pull request by erickson-doug**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 
### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).
### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] Validation errors from the [Linter extension for VS Code](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/linter.md) have all been fixed for this spec. (**Note:** for large, previously checked in specs, there will likely be many errors shown. Please contact our team so we can set a timeframe for fixing these errors if your PR is not going to address them).
  - [ ] The spec follows the patterns described in the [Swagger good patterns](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-good-patterns.md) document unless the service API makes this impossible.

---

This change is [Reviewable](https://reviewable.io/reviews/azure/azure-rest-api-specs/628)

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@erickson-doug__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (douge). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: documentation/swagger-authoring-descriptions.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+2. Previewing your content is a good practice. It takes you out of the curly-brace world of JSON, and lets you view your writing with a reader's eye, not a coder's. Empathy is important for great docs.
+
+3. Review the style guides used by your content partners:
+    - The all-up [Microsoft Writing Style Guide](https://worldready.cloudapp.net/StyleGuide/Read?id=2700)
+    - The Azure-specific [Microsoft Cloud Style Guide](https://worldready.cloudapp.net/StyleGuide/Read?id=2696)
+
+## Notes and Caveats
+
+1. Right now, description strings are very limited in their expressiveness. Our tooling does not support the rendering of GFM syntax, nor does it support any documentation-friendly vendor extensions at this time. (This will change.) Some HTML elements are whitelisted (permitted):
```
Comment (tbombach) [inline documentation/swagger-authoring-descriptions.md]: > Our tooling
Also, as Amar mentioned in the other PR, this is confusing for authors who are used to "our tooling" referring to AutoRest. Could you change this to "doc generation tooling?"

**Comment by tbombach** [comment]
Review status: 0 of 1 files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[documentation/swagger-authoring-descriptions.md, line 7 at r3](https://reviewable.io:443/reviews/azure/azure-rest-api-specs/628#-KUsyyQllTdU0Ie_Qjy1:-KUsyyQllTdU0Ie_Qjy2:beb5f8p) ([raw file](https://github.com/azure/azure-rest-api-specs/blob/a0d9da9199904b78b3ba2911d7579ed1c58f16ac/documentation/swagger-authoring-descriptions.md#L7)):*
> AutoREST
Minor: AutoRest (same goes for other usages in this doc)

---

*[documentation/swagger-authoring-descriptions.md, line 9 at r3](https://reviewable.io:443/reviews/azure/azure-rest-api-specs/628#-KUslPkPIlNZS6gTKzZE:-KUslPkPIlNZS6gTKzZF:byhzbw8) ([raw file](https://github.com/azure/azure-rest-api-specs/blob/a0d9da9199904b78b3ba2911d7579ed1c58f16ac/documentation/swagger-authoring-descriptions.md#L9)):*
> ```Markdown
> These descriptions are also passed through by AutoREST to become triple-slash (///) comments in the corresponding client SDKs, which are likewise transformed into published baseline ref documentation.
> 
> This means that these docs are part of your product, and as ScottGu notes, a key part of your user experience. They're not just served up on docs.microsoft.com; they can also be discovered through Swagger UI (or other REST documentation client tools), command-line tools, and IntelliSense (for client SDKs). These descriptions represent your product and the hard work you put into making great APIs, and as Microsoft, folks expect a polished experience.
> ```

Not sure if you're looking for wording feedback, but "as Microsoft" + "folks expect" doesn't sound quite right. Perhaps split the sentences?

---

*[documentation/swagger-authoring-descriptions.md, line 18 at r3](https://reviewable.io:443/reviews/azure/azure-rest-api-specs/628#-KUskh97T6ed_F5ihkrM:-KUskh97T6ed_F5ihkrN:bp7w62i) ([raw file](https://github.com/azure/azure-rest-api-specs/blob/a0d9da9199904b78b3ba2911d7579ed1c58f16ac/documentation/swagger-authoring-descriptions.md#L18)):*
> ```Markdown
> 2. Set yourself up with some good content support tools.
>     - [Grammarly](http://www.grammarly.com) is a web site that provides an English language grammar checker. Just cut-and-paste your description strings into it and review the suggested fixes.
>     - If you're using VSCode to edit your Swagger, consider one of these Swagger doc previewer extensions:
> ```

Could you link to the Swagger Linter extension that we provide as well: https://github.com/Azure/openapi-lint-extension#preview (this link goes directly to the screenshot of the feature)

---

*[documentation/swagger-authoring-descriptions.md, line 29 at r3](https://reviewable.io:443/reviews/azure/azure-rest-api-specs/628#-KUsmNbPHjAM04RjQVKX:-KUsmNbPHjAM04RjQVKY:bhv1ies) ([raw file](https://github.com/azure/azure-rest-api-specs/blob/a0d9da9199904b78b3ba2911d7579ed1c58f16ac/documentation/swagger-authoring-descriptions.md#L29)):*
> This would be a great enhancement to the Linter.
Could you open an issue in the https://github.com/Azure/openapi-lint-extension repo and we'll track it there. I think that would be better instead of having it part of this documentation

---

*[documentation/swagger-authoring-descriptions.md, line 30 at r3](https://reviewable.io:443/reviews/azure/azure-rest-api-specs/628#-KUsmjc1MMCTXvRYZYwW:-KUsmjc1MMCTXvRYZYwX:b-h4llcl) ([raw file](https://github.com/azure/azure-rest-api-specs/blob/a0d9da9199904b78b3ba2911d7579ed1c58f16ac/documentation/swagger-authoring-descriptions.md#L30)):*
> "not Url"
Minor: put not outside the quotes

---

*[documentation/swagger-authoring-descriptions.md, line 31 at r3](https://reviewable.io:443/reviews/azure/azure-rest-api-specs/628#-KUsn9vYxtN0fh9kjiYy:-KUsn9vYxtN0fh9kjiYz:b-huxy96) ([raw file](https://github.com/azure/azure-rest-api-specs/blob/a0d9da9199904b78b3ba2911d7579ed1c58f16ac/documentation/swagger-authoring-descriptions.md#L31)):*
> *documentid*
operationId?

---

*[documentation/swagger-authoring-descriptions.md, line 31 at r3](https://reviewable.io:443/reviews/azure/azure-rest-api-specs/628#-KUsz9QpOTmnLUa2n-v0:-KUsz9Qq4XgW7nJgnt0u:b-lxjudi) ([raw file](https://github.com/azure/azure-rest-api-specs/blob/a0d9da9199904b78b3ba2911d7579ed1c58f16ac/documentation/swagger-authoring-descriptions.md#L31)):*
> ("destroys)
Minor: ("destroys**"**)

---

*[documentation/swagger-authoring-descriptions.md, line 32 at r3](https://reviewable.io:443/reviews/azure/azure-rest-api-specs/628#-KUsnPImI7vrtUR0dKQG:-KUsnPImI7vrtUR0dKQH:bcafvu1) ([raw file](https://github.com/azure/azure-rest-api-specs/blob/a0d9da9199904b78b3ba2911d7579ed1c58f16ac/documentation/swagger-authoring-descriptions.md#L32)):*
> ```Markdown
> 3. **Do not use the API's code-symbol capitalization style for acronyms**. For example, use "URL", "not Url"; and use "ID", not "Id".
> 4. **Start the description for every operation with a verb phrase.** Try to use the same verb that you provided in the *documentid* (for AutoREST method naming) or for the HTTP operation (GET/PUT/DELETE, at least.) That said, some verbs are uncomfortable ("obtains") or unfriendly ("destroys) or may not make sense to a newer reader ("patches"), so put yourself in the reader's shoes (or look at similar, previously documented APIs) and choose a good, readable verb to open with. We prefer the imperative form ("Get...") over the third-person singular ("Gets..."), but that's mostly a grammar nerd dispute. 
> 5. **Document the HTTP status codes that your REST operations return**. Yeah, that's a pain, since you're probably thinking more about the client SDK APIs than the REST ones, but since we're publishing public docs for your REST APIs, it's pretty sloppy to have those fields empty -- and since someone **can** call these APIs through an HTTP interface directly, they will want to know what to do with those codes or what happened with their request. If you have good logic for handling these codes in your SDK implementations, share that briefly in the response status code descriptions.
> ```

One note: from a Swagger/AutoRest perspective, we do not currently encourage authors to provide non-positive status codes if the generated code should throw upon receiving that code. If 404 response is defined, then the code will not throw and the consumer will need to inspect the return object to determine the state of the returned item. The current behavior is to treat the `default` response as the schema for any non-described codes when deserializing the response

---

*[documentation/swagger-authoring-descriptions.md, line 47 at r3](https://reviewable.io:443/reviews/azure/azure-rest-api-specs/628#-KUsqROT0owz-Wc3IAXK:-KUsqROT0owz-Wc3IAXL:b5wsw2r) ([raw file](https://github.com/azure/azure-rest-api-specs/blob/a0d9da9199904b78b3ba2911d7579ed1c58f16ac/documentation/swagger-authoring-descriptions.md#L47)):*
> ```Markdown
> ## Notes and Caveats
> 
> 1. Right now, description strings are very limited in their expressiveness. Our tooling does not support the rendering of GFM syntax, nor does it support any documentation-friendly vendor extensions at this time. (This will change.) Some HTML elements are whitelisted (permitted):
> ```

I think we should not encourage HTML elements in descriptions for a number of reasons:
- Having any level of support for tags makes it difficult to validate a description. Some teams use <var_name> to call out variables: https://github.com/Azure/azure-rest-api-specs/blob/master/arm-cdn/2015-06-01/swagger/cdn.json#L1775. It's unclear whether this usage is the start of a tag or not without bringing in HTML parsing libraries
- GFM is already the supported markdown format according to Swagger, even though support for it will be added later for the doc tooling. Asking authors to put HTML to work around this point in time issue will create debt for ourselves when we want to get rid of those tags later.
- Other Swagger tools will often treat HTML as a security issue: https://github.com/swagger-api/swagger-ui/issues/830. I know it's most important for script tags (which aren't whitelisted), but it could cause issues for tools that safe encode the characters for output (which is what AutoRest wants to do in generated code)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/azure/azure-rest-api-specs/628)*

File: documentation/swagger-authoring-descriptions.md (hunk 1 above)
Comment (erickson-doug) [inline documentation/swagger-authoring-descriptions.md]: Will do.

**Pull request closed without merging**